File: 02. UI Testing/02. Selenium_PageObject/02. Java/02. WeareSocialNetwork/src/main/java/pages/LoginPage.java
package pages;

import org.openqa.selenium.WebDriver;

public class LoginPage extends BasePage {
    public LoginPage(WebDriver driver) {
        super(driver, "baseUrl");
    }

    public void authenticate(String username, String password) {
        actions.waitForElementVisibleUntilTimeout("signInPage.username", 30);
        actions.typeValueInField(username, "signInPage.username");
        actions.waitForElementVisible("signInPage.password", 30);
        actions.typeValueInField(password, "signInPage.password");
    }

    public void clickSignInButton() {
        actions.waitForElementVisibleUntilTimeout("signInPage.loginButton", 30);
        actions.clickElement("signInPage.loginButton");
    }
}

File: 02. UI Testing/02. Selenium_PageObject/01. C#/1.httpsautomationexercise.com/AutomationExercise.Tests/Pages/LoginPage.cs
using OpenQA.Selenium;
using AutomationExercise.Tests.Selectors;

namespace AutomationExercise.Tests.Pages
{
    public class LoginPage : BasePage
    {
        public LoginPage(IWebDriver driver) : base(driver) { }

        public bool IsLoginHeaderVisible()
        {
            return IsElementDisplayed(LoginPageSelectors.LoginHeader);
        }

        public HomePage Login(string email, string password)
        {
            EnterText(LoginPageSelectors.LoginEmailInput, email);
            EnterText(LoginPageSelectors.LoginPasswordInput, password);
            Click(LoginPageSelectors.LoginButton);
            return new HomePage(Driver);
        }

        public string GetLoginErrorText()
        {
            return GetElementText(LoginPageSelectors.LoginErrorText);
        }

        public bool IsLoginErrorVisible()
        {
            return IsElementDisplayed(LoginPageSelectors.LoginErrorText);
        }

        public bool IsSignupHeaderVisible()
        {
            return IsElementDisplayed(LoginPageSelectors.SignupHeader);
        }

        public SignupPage Signup(string name, string email)
        {
            EnterText(LoginPageSelectors.SignupNameInput, name);
            EnterText(LoginPageSelectors.SignupEmailInput, email);
            Click(LoginPageSelectors.SignupButton);
            return new SignupPage(Driver);
        }

        public string GetSignupErrorText()
        {
            return GetElementText(LoginPageSelectors.SignupErrorText);
        }

        public bool IsSignupErrorVisible()
        {
            return IsElementDisplayed(LoginPageSelectors.SignupErrorText);
        }
    }
}
